pkg/services/schemaloader/schemaloader.go

```diff
@@ -147,3 +153,101 @@ func removeNilArray(initialArray []interface{}) []interface{} {
 	}
 	return withoutNils
 }
+
+type file struct {
+	name    string
+	content *bytes.Buffer
+	modTime time.Time
+	closed  bool
+}
+
+func (f *file) Read(p []byte) (int, error) {
+	if f.closed {
+		return 0, errors.New("file closed")
+	}
+	return f.content.Read(p)
+}
+
+func (f *file) Stat() (fs.FileInfo, error) {
+	if f.closed {
+		return nil, errors.New("file closed")
+	}
+	return f, nil
+}
+
+func (f *file) Close() error {
+	f.closed = true
+	return nil
+}
+
+func (f *file) Name() string {
+	return f.name
+}
+
+func (f *file) Size() int64 {
+	return int64(f.content.Len())
+}
+
+func (f *file) Mode() fs.FileMode {
+	return 0444
+}
+
+func (f *file) ModTime() time.Time {
+	return f.modTime
+}
+
+func (f *file) IsDir() bool {
+	return false
+}
+
+func (f *file) Sys() interface{} {
+	return nil
+}
+
+type InstanceFS struct {
+	files map[string]*file
+}
+
+func NewInstanceFS() *InstanceFS {
+	return &InstanceFS{
+		files: make(map[string]*file),
```

Comment:
this is probably not enough, regarding if we want to reuse https://github.com/grafana/grafana/blob/main/pkg/schema/load/panel.go#L72, need to have directory instead only plain files